### PR TITLE
Add a Settings UI Prototype

### PR DESCRIPTION
**Pull request by carlos-zamora**
This commit introduces a rough prototype of the Settings UI as the
`TerminalSettingsEditor`. This project was added to OpenConsole.sln and
deploys as a standalone app. Some databinding is configured to a fake
TerminalSettingsModel (located in the ObjectModel folder).

This commit will start the settings UI feature branch, which will
receive a full review on merge-back into main.

References #6720 - Settings UI Spec and Design
References #1564 - Settings UI Feature/Epic

### CODE REVIEW

File: src/cascadia/TerminalControl/desktop.ini
Comment (carlos-zamora) [reply src/cascadia/TerminalControl/desktop.ini:1]: that was an accident haha. Thanks!

File: src/cascadia/TerminalSettingsEditor/GlobalSettings.idl
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/GlobalSettings.idl:27]: Bare TODO

File: src/cascadia/TerminalSettingsEditor/Globals.xaml
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/Globals.xaml:59]: Can you set some sort of style property or re-usable margin for all of these? It's weird to be applying "0,0,0,10" to everything over and over.

File: src/cascadia/TerminalSettingsEditor/Keybindings.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/Keybindings.cpp:37]: Dead code?

File: src/cascadia/TerminalSettingsEditor/Keybindings.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/Keybindings.cpp:204]: Is it necessary to assign the empty `wchar_t*` to `hstring` to get a basic, empty allocation, or would the default constructor suffice? (Here and below.)

File: src/cascadia/TerminalSettingsEditor/MainPage.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/MainPage.cpp:111]: Not sure if this is the only place, but we generally don't do single line if statements without braces...

File: src/cascadia/TerminalSettingsEditor/ObjectModel/AppSettings.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/AppSettings.cpp:93]: Uh, what? unused file now?

File: src/cascadia/TerminalSettingsEditor/ObjectModel/AppSettings.h
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/AppSettings.h:64]: Leftovers?

File: src/cascadia/TerminalSettingsEditor/ObjectModel/ColorScheme.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/ColorScheme.cpp:5]: These aren't in your PCH?

File: src/cascadia/TerminalSettingsEditor/ObjectModel/ColorScheme.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/ColorScheme.cpp:71]: Bare TODO

Comment (miniksa) [reply src/cascadia/TerminalSettingsEditor/ObjectModel/ColorScheme.cpp:71]: (and below)

File: src/cascadia/TerminalSettingsEditor/ObjectModel/ColorScheme.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/ColorScheme.cpp:358]: I mean at that point, you should also make it a helper instead of copypasta down the whole file.

File: src/cascadia/TerminalSettingsEditor/ObjectModel/GlobalSettings.idl
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/GlobalSettings.idl:39]: Bare TODO

File: src/cascadia/TerminalSettingsEditor/ObjectModel/Profile.h
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/Profile.h:36]: Shouldn't some of these defaults load out of `defaults.json` instead of having another matching definition of them here?

File: src/cascadia/TerminalSettingsEditor/ObjectModel/Profile.idl
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/Profile.idl:101]: This TODO should probably become a GH TODO.

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/Profiles.cpp:56]: Batch files and EXE files is an interesting choice for command lines. There are other executable types and scripts...

File: src/cascadia/TerminalSettingsEditor/TerminalSettingsEditor.vcxproj
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/TerminalSettingsEditor.vcxproj:46]: Might want to drop some of these things and include our common props files for cascadia and/or the entire project.

**Review by miniksa — DISMISSED** [review]

**Comment by carlos-zamora** [comment]
I resolved most of the ObjectModel comments without actually doing anything. These files are temporary. After this PR goes in, they'll be deleted and we'll actually hook up the Terminal Settings Model objects instead.

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/Profiles.cpp:56]: So, this one's weird. It's still a textbox so the user should be able to put in anything they want. That said, what other file extensions are supported here? I can just add them to the list.

**Comment by carlos-zamora** [comment]
Could I get a bit of guidance for how to handle these misspellings? Most seem to be function names.

**Comment by zadjii-msft** [comment]
without looking at any code of course

* addnew - looks like a real typo
* bright* - I'd make these camelCase if possible, that'll fix the bot
* delimeters - looks like a real typo
* globalprofile - can you camelCase this?
* homegriditem - can you camelCase this?
* muxc - microsoft.txt
* snprintf - apis.txt
* Subheader - can you camelCase this?
* Subpage - can you camelCase this?
* tolower - apis.txt
* viceversa - "vice-versa"
* wcsstr - apis.txt

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp
Comment (htcfreek) [reply src/cascadia/TerminalSettingsEditor/Profiles.cpp:56]: What about .cmd files?

Comment (zadjii-msft) [reply src/cascadia/TerminalSettingsEditor/Profiles.cpp:56]: Don't do `bat` or `cmd` fies - not unless we're prepared to auto-prefix them with a `cmd.exe /c `. You can't just call `CreateProcess` on a `bat` file, it needs to be executable.

File: src/cascadia/TerminalSettingsEditor/ObjectModel/GlobalSettings.idl
Comment (miniksa) [inline src/cascadia/TerminalSettingsEditor/ObjectModel/GlobalSettings.idl:39]: shouldn't this still have a real todo somewhere

File: src/cascadia/TerminalSettingsEditor/Globals.xaml
Comment (miniksa) [reply src/cascadia/TerminalSettingsEditor/Globals.xaml:59]: eh its fine.

File: src/cascadia/TerminalSettingsEditor/ObjectModel/GlobalSettings.idl
Comment (carlos-zamora) [reply src/cascadia/TerminalSettingsEditor/ObjectModel/GlobalSettings.idl:39]: That's in code that's getting deleted but removed it anyways.

File: src/cascadia/TerminalSettingsEditor/ColorSchemes.xaml
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/ColorSchemes.xaml:268]: NOTE FOR EPIC: REFACTOR button color hex flyouts!

Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/ColorSchemes.xaml:268]: They should be one control. Not 150 copies of the same control.

File: src/cascadia/TerminalSettingsEditor/GlobalAppearance.xaml
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/GlobalAppearance.xaml:3]: NOTE FOR EPIC: Make sure *all strings* are attached to UIDs.

File: src/cascadia/TerminalSettingsEditor/KeyBindingsPage.xaml
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/KeyBindingsPage.xaml:14]: hope you didn't add asdf to the spellchecker for this

File: src/cascadia/TerminalSettingsEditor/KeyBindingsPage.xaml.h
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/KeyBindingsPage.xaml.h:16]: hold up, kill this file and the other .xaml.cpp and .xaml.h files.

Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/KeyBindingsPage.xaml.h:16]: they are not C++ files. they do not belong in our C++ project.

File: src/cascadia/TerminalSettingsEditor/ObjectModel/AppSettings.cpp
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/ObjectModel/AppSettings.cpp:95]: fake settings model. roll with it til we delete it.

File: src/cascadia/TerminalSettingsEditor/Package.appxmanifest
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Package.appxmanifest:13]: LEON TO DELETE

File: src/cascadia/TerminalSettingsEditor/Profiles.cpp
Comment (DHowett) [reply src/cascadia/TerminalSettingsEditor/Profiles.cpp:56]: Not blocking on this. This is going into a feature branch that will have a final review when it lands.


NOTE FOR EPIC: **MAKE SURE COMMANDLINE HANDLING IS ROBUST AND INTELLIGENT**

File: src/cascadia/TerminalSettingsEditor/Profiles.xaml
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Profiles.xaml:109]: obvs you know you can't do this

File: src/cascadia/TerminalSettingsEditor/PropertySheet.props
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/PropertySheet.props:14]: LEON TO DELETE

File: src/cascadia/TerminalSettingsEditor/Utils.cpp
Comment (DHowett) [inline src/cascadia/TerminalSettingsEditor/Utils.cpp:26]: namespace these.

**Review by DHowett — APPROVED** [review]
### Rationale

This is merging into a feature branch that Leon and Carlos are going to be hammering on for a while. It is a prototype, yes, but it is getting refined.

There will be a final review when the feature branch merges.

File: src/cascadia/TerminalSettingsEditor/Keybindings.cpp
Comment (leonMSFT) [inline src/cascadia/TerminalSettingsEditor/Keybindings.cpp:200]: We could probably use the `fmt` lib to replace all the string building - here and everywhere else?

File: src/cascadia/TerminalSettingsEditor/Launch.xaml
Comment (leonMSFT) [inline src/cascadia/TerminalSettingsEditor/Launch.xaml:27]: We're prob gonna have to move these tooltips into our `Resources.resw`

**Review by leonMSFT — CHANGES_REQUESTED** [review]
TBH I think a lot of the things I saw are stuff that's going to be deleted and polished naturally as we work towards endgame. i.e. large amounts of repeating XAML elements just becomes a templated item + data binding.

Blocking for now just so we can sprinkle TODOs throughout the code first before merging this in.

File: src/cascadia/TerminalSettingsEditor/KeyBindingsPage.xaml
Comment (leonMSFT) [reply src/cascadia/TerminalSettingsEditor/KeyBindingsPage.xaml:14]: image
funny enough, it already exists in `expect.txt` 😆

File: src/cascadia/TerminalSettingsEditor/Keybindings.cpp
Comment (leonMSFT) [reply src/cascadia/TerminalSettingsEditor/Keybindings.cpp:200]: For now I'll put a TODO and a reminder in the epic to do this as we go, I expect a lot of the strings in the proj to get changed.

**Review by leonMSFT — APPROVED** [review]
For the Leon callouts - those changes are currently already in the branch that I'm currently working on to put the settings UI in a tab, so they'll come in a later PR.

I've also put the Epic callouts in a follow up list in #1564. I expect we'll be referring back to that list often as we work thoroughly in the code.

**Pull request merged**
